airflow/models.py

```diff
@@ -1238,7 +1242,11 @@ def run(
             # todo: move this to the scheduler
                 self.state == State.UP_FOR_RETRY and
                 not self.ready_for_retry()):
-            next_run = (self.end_date + task.retry_delay).isoformat()
+            delay = task.retry_delay
```

Comment:
This code is copy/paste of the above. A helper function seems like it'd be good.